text/0156-columnar-encoding.md

```diff
@@ -0,0 +1,1389 @@
+
+* for each Arrow Schema create a list of triples (name, type, metadata) for each column.
+* sort these triples according to a lexicographic order.
+* concatenate the sorted triples with a separator and use these identifiers as `schema_id` (or a shorter version via
+  an equivalence table).
+
+### Traffic Balancing Optimization
+
+To mitigate the usual pitfalls of a stream-oriented protocol, protocol implementers are advised to:
+
+* client side: create several streams in parallel (e.g. create a new stream every 10 event types),
+* server side: close streams that have been open for a long time (e.g. close stream every 1 hour).
+
+These parameters must be exposed in a configuration file and be tuned according to the application.
+
+### Throttling
+
+OTLP Arrow allows backpressure signaling. If the server is unable to keep up with the pace of data it receives from the
+client then it should signal that fact to the client. The client must then throttle itself to avoid overwhelming the
+server.
+
+To signal backpressure when using OTLP Arrow, the server should return an error with code UNAVAILABLE and may supply
+additional details via the `retry_info` attribute.
+
+When the client receives this signal it should follow the recommendations outlined in documentation for RetryInfo:
+
+```
+// Describes when the clients can retry a failed request. Clients could ignore
+// the recommendation here or retry when this information is missing from error
+// responses.
+//
+// It's always recommended that clients should use exponential backoff when
+// retrying.
+//
+// Clients should wait until `retry_delay` amount of time has passed since
+// receiving the error response before retrying. If retrying requests also
+// fail, clients should use an exponential backoff scheme to gradually increase
+// the delay between retries based on `retry_delay`, until either a maximum
+// number of retires have been reached or a maximum retry delay cap has been
+// reached.
+```
+
+The value of retry_delay is determined by the server and is implementation dependant. The server should choose a
+retry_delay value that is big enough to give the server time to recover, yet is not too big to cause the client to drop
+data while it is throttled.
+
+Throttling is important for implementing reliable multi-hop telemetry data delivery all the way from the source to the
+destination via intermediate nodes, each having different processing capacity and thus requiring different data transfer
+rates.
+
+### Best Effort Delivery Guarantee
+
+TBD (phase 1)
+
+## Risks and Mitigations
+
+An authentication mechanism is highly recommended to protect against malicious traffic. Without authentication, an OTLP
+receiver can be attacked in multiple ways ranging from DoS, traffic amplification to sending sensitive data. This
+specification reuses the authentication mechanisms already in place in the collector.
```

Comment:
hmmm.... arrow Flight already has a Handshake and other authentication options built-in that could be leveraged.....

just saying :smile: lol

Reply:
Yes indeed, but see my [previous answer](https://github.com/open-telemetry/oteps/pull/171#discussion_r1069808903) explaining the motivations behind this choice :)